extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -80,47 +80,43 @@ private CosmosDBInputAttribute GetBindingDataContent(ModelBindingData bindingDat
             Type _ when targetType == typeof(CosmosClient) => CreateCosmosClient<CosmosClient>(cosmosAttribute),
             Type _ when targetType == typeof(Database) => CreateCosmosClient<Database>(cosmosAttribute),
             Type _ when targetType == typeof(Container) => CreateCosmosClient<Container>(cosmosAttribute),
-            _ => await CreateTargetObjectAsync(targetType, cosmosAttribute)
+            _ => ConvertDynamic(await CreateTargetObjectAsync(targetType, cosmosAttribute), targetType)
         };
 
-        private async Task<object> CreateTargetObjectAsync(Type targetType, CosmosDBInputAttribute cosmosAttribute)
+        private object ConvertDynamic(dynamic data, Type targetType)
         {
-            MethodInfo createPOCOMethod;
-
-            if (targetType.GenericTypeArguments.Any())
-            {
-                targetType = targetType.GenericTypeArguments.FirstOrDefault();
-
-                createPOCOMethod = GetType()
-                                    .GetMethod(nameof(CreatePOCOCollectionAsync), BindingFlags.Instance | BindingFlags.NonPublic)
-                                    .MakeGenericMethod(targetType);
-            }
-            else
-            {
-                createPOCOMethod = GetType()
-                                    .GetMethod(nameof(CreatePOCOAsync), BindingFlags.Instance | BindingFlags.NonPublic)
-                                    .MakeGenericMethod(targetType);
-            }
-
+            var json = JsonSerializer.Serialize(data);
+            return JsonSerializer.Deserialize(json, targetType,
+                new JsonSerializerOptions() { PropertyNameCaseInsensitive = true });
```

Comment:
Use a cached/static version of `JsonSerializerOptions` instance instead of creating one in every Convert call.